src/site/antora/modules/ROOT/pages/manual/thread-context.adoc

```diff
@@ -159,7 +159,20 @@ You can provide a custom `ContextDataProvider` implementation as follows:
 ----
 META-INF/services/org.apache.logging.log4j.core.util.ContextDataProvider
 ----
-. Write the fully-qualified class name of your custom implementation (e.g., `com.mycompany.CustomContextDataProvider`) to the file created in the previous step
+. Write the fully-qualified class name of your custom implementation (e.g., `com.mycompany.CustomContextDataProvider`) to the file created in the previous step.
+
+==== Third-party custom thread context data providers
+
+[IMPORTANT]
+====
+These context data providers are supplied by **third-party** vendors and are not maintained by the Apache Software Foundation.
```

Comment:
```suggestion
These lookups are provided by **third-party** vendors and are not maintained by the link:{logging-services-url}[Apache Logging Services] (Log4j, Log4cxx, Log4net) project.
```

Reply:
Fixed in link:{logging-services-url}[Apache Logging Services] (Log4j, Log4cxx, Log4net) project